task/dump-agent/src/udp.rs

```diff
@@ -0,0 +1,167 @@
+                }
+            };
+
+        if let Some(out_len) = out_len {
+            meta.size = out_len as u32;
+            if let Err(e) = self.net.send_packet(
+                SocketName::dump_agent,
+                meta,
+                &tx_data_buf[..meta.size as usize],
+            ) {
+                // We'll drop packets if the outgoing queue is full;
+                // the host is responsible for retrying.
+                //
+                // Other errors are unexpected and panic.
+                //
+                // This includes ServerRestarted, because the server should only
+                // restart if the watchdog times out, and the watchdog should
+                // not be timing out, because we're literally replying to a
+                // packet here.
+                ringbuf_entry!(Trace::SendError(e));
+                match e {
+                    SendError::QueueFull => (),
+                    SendError::Other
+                    | SendError::ServerRestarted
+                    | SendError::NotYours
+                    | SendError::InvalidVLan => panic!(),
+                }
+            }
+        }
+    }
+
+    /// Handles a single message
+    fn handle_message(
+        &mut self,
+        header: humpty::udp::Header,
+        data: &[u8],
+    ) -> Result<humpty::udp::Response, humpty::udp::Error> {
+        // If the header is < our min version, then we can't deserialize at all,
+        // so return an error immediately.
+        if header.version < humpty::udp::version::MIN {
+            ringbuf_entry!(Trace::WrongVersion(header.version));
+            return Err(humpty::udp::Error::VersionMismatch);
+        }
+
+        use humpty::udp::{Request, Response};
+        let r = match hubpack::deserialize::<Request>(data) {
+            Ok((msg, _data)) => match msg {
+                Request::ReadDump { index, offset } => {
+                    self.read_dump(index, offset).map(Response::ReadDump)?
+                }
+                Request::GetDumpArea { index } => {
+                    self.dump_area(index).map(Response::GetDumpArea)?
+                }
+                Request::InitializeDump => {
+                    self.initialize().map(|()| Response::InitializeDump)?
+                }
+                Request::AddDumpSegment { address, length } => self
+                    .add_dump_segment(address, length)
+                    .map(|()| Response::AddDumpSegment)?,
+                Request::TakeDump => {
+                    self.take_dump().map(|()| Response::TakeDump)?
+                }
+            },
+            Err(e) => {
+                // This message is from a newer version, so it makes sense that
+                // we failed to deserialize it.
+                if header.version > humpty::udp::version::CURRENT {
+                    ringbuf_entry!(Trace::WrongVersion(header.version));
+                    return Err(humpty::udp::Error::VersionMismatch);
```

Comment:
Could we include the SP's `version::CURRENT` (and probably `header.version` while we're at it) in this error variant?

Reply:
Yup, done in 912a2674dd7b243c521f3011702689e8ee64157e